src/js/buildStandalone.js

```diff
@@ -19,56 +19,56 @@ clientStandaloneRequest.onreadystatechange = function() {
 }
 clientStandaloneRequest.send();
 
+// replace bad characters in user-supplied text
+function safeUser(value) {
+	const regex = /[<>&"'\v]/g;
+	var nv = value;
+	if (nv.match(regex)) {
+		consolePrint(`Unsafe characters found in script will be replaced by !`);
+		nv = nv.replace(regex, '!');
+	}
+	// remove $ for now, they go back later
+	return safeDollar(nv);
+}
+
+// replace $ because it's special in replace
+function safeDollar(value) {
+	return value.replace(/[$]/g, '\v');
+}
+
+// mainline function to build and save standalone version of script
 function buildStandalone(sourceCode) {
 	if (standalone_HTML_String.length===0) {
 		consolePrint("Can't export yet - still downloading html template.",true,null,null);
 		return;
 	}
 
 	var htmlString = standalone_HTML_String.concat("");
-	var title = "PuzzleScript Game";
-	if (state.metadata.title!==undefined) {
-		title=state.metadata.title;
-	}
-	var homepage = "https://www.puzzlescript.net";
-	if (state.metadata.homepage!==undefined) {
-		homepage=state.metadata.homepage;
-		if (!homepage.match(/^https?:\/\//)) {
-			homepage = "https://" + homepage;
-		}
-	}
-	var homepage_stripped = homepage.replace(/^https?:\/\//,'');
-
-	var background_color="black";
-	if ('background_color' in state.metadata) {
-		background_color=state.bgcolor;		
-	}
-	htmlString = htmlString.replace(/___BGCOLOR___/g,background_color);	
+	var title = state.metadata.title ? state.metadata.title : "PuzzleScript Game";
```

Comment:
In JavaScript `null`s are typically coalesced like:
```javascript
var title = state.metadata.title || "PuzzleScript Game";
```

Reply:
That's my C/C++/C# showing up! I do know the JS way, but in so much of this code this would be written as an if/else with braces, I thought I was going out on a limb just making it a one-liner.  But I approve the change.